src/types/utils.ts

```diff
@@ -53,6 +53,11 @@ export type DeepPartialSkipArrayKey<T> = T extends
   ? { [K in keyof T]: DeepPartialSkipArrayKey<T[K]> }
   : { [K in keyof T]?: DeepPartialSkipArrayKey<T[K]> };
 
+export type UnPackDefaultValues<TFieldValues> =
+  TFieldValues extends () => Promise<any>
+    ? Awaited<ReturnType<TFieldValues>>
+    : TFieldValues;
```

Comment:
I think this can be shortened.
```suggestion
  TFieldValues extends () => Promise<infer U> ? U : TFieldValues;
```